react/src/components/login.js

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+import { Link, Redirect } from 'react-router-dom';
+
+import { LOGIN } from '../utilities/constants';
+import { logInUser } from '../actions/index';
+
+import '../css/login.css';
+
+const LogIn = props => {
+  const [credentials, setCredentials] = useState(LOGIN);
+  const { token, logInUser } = props;
+
+  const handleChange = ({ target: { name, value } }) => {
+    switch (name) {
+      case 'email':
+      case 'password':
+        setCredentials({ ...credentials, [name]: value });
+        break;
+      default:
+        break;
+    }
```

Comment:
There's no need for the fall-through switch now, since there's only one generic state updateder function that can update all the state values:


```
const handleChange = ({target: { name, value } }) =>  {
    setCredentials({ ...credentials, [name]: value });
}
```

Reply:
True, I was looking afterward, where was the other component, I had inserted that, slipped through my eyes. Thanks, by the way, an elegant way to solve it without the switch block.